tests/test_tokenization_fast.py

```diff
@@ -258,281 +254,275 @@ def assert_build_inputs_with_special_tokens(self, tokenizer_r, tokenizer_p):
+            ((3, 8), "naive"),  # BERT normalizes this away
+            # Append MASK here after lower-casing
+            ((16, 21), "Allen"),
+            ((22, 24), "##NL"),
+            ((24, 25), "##P"),
+            ((26, 34), "sentence"),
+            ((35, 36), "."),
+        ]
 
-            # Check we have the same number of added_tokens for both pair and non-pair inputs.
-            self.assertEqual(tokenizer_r.num_added_tokens(False), tokenizer_p.num_added_tokens(False))
-            self.assertEqual(tokenizer_r.num_added_tokens(True), tokenizer_p.num_added_tokens(True))
+        # Check if the tokenizer is uncased
+        if tokenizer_r.init_kwargs.get("do_lower_case"):
+            expected_results = [(offset, token.lower()) for (offset, token) in expected_results]
 
-            # Check we have the correct max_length for both pair and non-pair inputs.
-            self.assertEqual(tokenizer_r.max_len_single_sentence, tokenizer_p.max_len_single_sentence)
-            self.assertEqual(tokenizer_r.max_len_sentences_pair, tokenizer_p.max_len_sentences_pair)
+        # Append the special tokens
+        expected_results.insert(3, ((9, 15), "[MASK]"))
+        expected_results.insert(0, (None, "[CLS]"))
+        expected_results.append((None, "[SEP]"))
 
-            # Assert the set of special tokens match.
-            self.assertSequenceEqual(
-                tokenizer_p.special_tokens_map.items(),
-                tokenizer_r.special_tokens_map.items(),
-                "Roberta tokenizers doesn't have the same set of special_tokens",
-            )
+        self.assertEqual([e[1] for e in expected_results], tokenizer_r.convert_ids_to_tokens(tokens["input_ids"]))
+        # self.assertEqual([e[0] for e in expected_results], tokens["offset_mapping"])
 
-            # Assure tokenization overlap between python and rust impl.
-            self.assert_tokenization_python_rust_almost_equals(tokenizer_p, tokenizer_r, 0.01)
 
-            # Ensure add_tokens and add_special_tokens return the correct vocab size
-            self.assert_add_tokens(tokenizer_r)
+class RobertaFastTokenizerTest(CommonFastTokenizerTest):
+    TOKENIZERS_CLASSES = frozenset(
+        [Tokenizer("Roberta", RobertaTokenizerFast, RobertaTokenizer, "vocab_file", filter_roberta_detectors),]
+    )
 
-            # Check for offsets mapping
-            self.assert_offsets_mapping(tokenizer_r)
+    def assert_embeded_special_tokens(self, tokenizer_r, tokenizer_p):
+        sentence = "A, <mask> AllenNLP sentence."
+        tokens_r = tokenizer_r.encode_plus(sentence, add_special_tokens=True, return_token_type_ids=True)
+        tokens_p = tokenizer_p.encode_plus(sentence, add_special_tokens=True, return_token_type_ids=True)
 
-            # Check for dynamic encoding sequence handling in batch_encode_plus
-            self.assert_batch_encode_dynamic_overflowing(tokenizer_r)
+        # Rust correctly handles the space before the mask while python doesnt
```

Comment:
I think Python is right here. There shouldn't be any space before the `<mask>` token. This means that Roberta on the fast path should probably have an `AddedToken('<mask>', lstrip=True)`